### ISSUE TITLE
Question: Role of Examples and Packages  folders,

### ISSUE THREAD

**Issue by ramandhingra** [NiGhTTraX/ts-monorepo#76]
I do not know much about lerna. But I have seen that the packages directory typically consists of sub-directories whose names are something like: frontend (or web), backend (or api), shared.)

Could you pl. answer these:

a. What exactly is the role of foo, bar, and components directories residing in the packages directory.
b. Why do the cra and nest appear in the examples directory.

Thanks

**Comment by ramandhingra** [comment]
Looks like the 'packages' directory contain the components and the logic that can be shared between the 'examples' like cra, next, and nestjs. 

So this probably means that in a real-world application, we should put the 'cra' (web sub-directory) and 'nestjs' (api sub-folder) inside the parent folder whose name could be packages. 

While, a better name for the 'packages' might be 'shared'. Am I right?

**Comment by NiGhTTraX** [comment]
Hey @ramandhingra, `packages` is conventionally the folder that keeps all of the different projects inside a monorepo. You can have more than one such folder and the names are for you to choose and configure through [pkg.workspaces](https://github.com/NiGhTTraX/ts-monorepo/blob/1000b559b8d0b8f6faf2cf6455910abeff7fcbb8/package.json#L5-L8).

In this repo, I have 2 folders, `packages` and `examples`. `packages` contain simpler projects that don't depend on any framework, and are meant to illustrate publishing to npm from within a monorepo. `examples` contain projects that use a framework and need a bit of config to work inside a monorepo. In a typical repo you could have all of them inside 1 folder, but here I wanted a clear separation between them for educational purposes.

TL;DR: the folders only mean something if you want them to mean something. Putting everything in one place is perfectly fine.

Hope that answers your question!

**Comment by ramandhingra** [comment]
Great. The answer is perfect. Thanks.

**Comment by NiGhTTraX** [comment]
@ramandhingra the [npm branch](https://github.com/NiGhTTraX/ts-monorepo/tree/npm) is set up with npm 7 :) The only difference is switching `yarn` for `npm` in commands.
